shopify/loaders/getAddresses.ts

```diff
@@ -0,0 +1,50 @@
+import { PostalAddress } from "../../commerce/types.ts";
+import type { AppContext } from "../mod.ts";
+import { FetchCustomerAddresses } from "../utils/storefront/queries.ts";
+import { Customer } from "../utils/storefront/storefront.graphql.gen.ts";
+import { getUserCookie } from "../utils/user.ts";
+
+/**
+ * @title Shopify Integration
+ * @description Get User Addresses
+ */
+const loader = async (
+  _props: unknown,
+  req: Request,
+  ctx: AppContext,
+): Promise<PostalAddress[] | null> => {
+  const { storefront } = ctx;
+
+  const customerAccessToken = getUserCookie(req.headers);
+
+  if (!customerAccessToken) return null;
+
+  try {
+    const customer = await storefront.query<
+      { customer: Customer },
+      { customerAccessToken: string }
+    >({
+      variables: { customerAccessToken },
+      ...FetchCustomerAddresses,
+    }).then((data) => data.customer);
+
+    if (!customer) return null;
+
+    const addresses = customer.addresses.edges;
+
+    return addresses.map((address) => ({
+        "@type": "PostalAddress",
+        "@id": address.node.id,
```

Comment:
```suggestion
        "identifier": address.node.id,
```

the "@id" should be used to reference this object inside a schema, for the actual id of the object maybe the "identifier" makes more sense